helmfile.d/charts/log-manager/files/compaction.sh

```diff
@@ -1,11 +1,17 @@
 #!/usr/bin/env bash
 
 set -euo pipefail
-
+if [ "$STORAGE_SERVICE" = "azure" ]; then
+# Azure Blob configuration
+: "${AZURE_STORAGE_CONNECTION_STRING:?Missing AZURE_STORAGE_CONNECTION_STRING}"
+: "${AZURE_CONTAINER_NAME:?Missing AZURE_CONTAINER_NAME}"
+: "${AZURE_PREFIX:?Missing AZURE_PREFIX}"
+else
+# S3 configuration
```

Comment:
```suggestion

if [ "$STORAGE_SERVICE" = "azure" ]; then
  # Azure Blob configuration
  : "${AZURE_STORAGE_CONNECTION_STRING:?Missing AZURE_STORAGE_CONNECTION_STRING}"
  : "${AZURE_CONTAINER_NAME:?Missing AZURE_CONTAINER_NAME}"
  : "${AZURE_PREFIX:?Missing AZURE_PREFIX}"
else
  # S3 configuration
```
And indent the S3 section as well.